### PR TITLE
BehekaHW_6.1

### PR DESCRIPTION
**Pull request by MishaBeheka**

### CODE REVIEW

File: src/main/java/core/basesyntax/Pair.java [hunk 1]
```diff
@@ -1,5 +1,45 @@
...
+        result = prime + result * first.hashCode();
+        result = prime + result * second.hashCode();
+        return result;
+    }
+
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (this.getClass() != obj.getClass() || obj == null) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: what will happen if obj is null?

File: src/main/java/core/basesyntax/Pair.java [hunk 2]
```diff
@@ -1,5 +1,45 @@
 package core.basesyntax;
 
-class Pair {
+class Pair<F, S> {
+
+    public static <F, S> Pair of(F first, S second) {
+        return new Pair<>(first, second);
+    }
+
+    F first;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: add access modifier

File: src/main/java/core/basesyntax/Pair.java [hunk 3]
```diff
@@ -1,5 +1,45 @@
...
+    S second;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Pair.java]: same

**Review by Tarabarnik — COMMENTED** [review]
Important!


Recommended order of class contents is:

Static fields.
Non-static fields.
Constructors.
Public methods.
Other methods.

File: src/main/java/core/basesyntax/Pair.java (hunk 1 above)
Comment (MishaBeheka) [reply src/main/java/core/basesyntax/Pair.java]: I will get false. I deleted this check

Comment (Tarabarnik) [reply src/main/java/core/basesyntax/Pair.java]: Nope. You'll get NullPointerException as your getClass check goes first. put them in correct order as in 3.2 hw and I'll approve

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**